aws_quickstart/datadog_integration_api_call_v2.yaml

```diff
@@ -0,0 +1,220 @@
+      and do not want to use Datadog Infrastructure Monitoring.
+    Default: false
+  CloudSecurityPostureManagement:
+    Type: String
+    AllowedValues:
+      - true
+      - false
+    Description: >-
+      Add the AWS Managed SecurityAudit policy to your Datadog AWS Integration role, and enable 
+      Datadog Cloud Security Posture Management (CSPM) to start performing configuration checks across your AWS account.
+      Datadog CSPM is a product that automatically detects resource misconfigurations in your AWS account according to 
+      industry benchmarks. More info: https://www.datadoghq.com/product/security-platform/cloud-security-posture-management/
+    Default: false
+Resources:
+  LambdaExecutionRoleDatadogAPICall:
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service:
+                - lambda.amazonaws.com
+            Action:
+              - sts:AssumeRole
+      Path: "/"
+      ManagedPolicyArns:
+        - !Sub "arn:${AWS::Partition}:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole"
+  # Retrieving secrets passed in via SecretsManager Arn
+  DatadogAPICall:
+    Type: "Custom::DatadogAPICall"
+    Properties:
+      ServiceToken: !GetAtt DatadogAPICallFunction.Arn
+      APIKey: !Ref DatadogApiKey
+      APPKey: !Ref DatadogAppKey
+      ApiURL: !Ref DatadogSite
+      AccountId: !Ref AWS::AccountId
+      RoleName: !Ref RoleName
+      HostTags: [!Sub "aws_account:${AWS::AccountId}"]
+      CloudSecurityPostureManagement: !Ref CloudSecurityPostureManagement
+      DisableMetricCollection: !Ref DisableMetricCollection
+  DatadogAPICallFunction:
+    Type: "AWS::Lambda::Function"
+    Properties:
+      Description: "A function to call the Datadog API."
+      Role: !GetAtt LambdaExecutionRoleDatadogAPICall.Arn
+      Handler: "index.handler"
+      Runtime: "python3.8"
+      Timeout: 30
+      Code:
+        ZipFile: |
+          import boto3
+          
+          import json
+          import logging
+          import signal
+          from urllib.request import build_opener, HTTPHandler, Request
+          import urllib.parse
+          
+          LOGGER = logging.getLogger()
+          LOGGER.setLevel(logging.INFO)
+          
+          def call_datadog_api(event, method):
+              api_key = event['ResourceProperties']['APIKey']
+              app_key = event['ResourceProperties']['APPKey']
+              api_url = event['ResourceProperties']['ApiURL']
+              account_id = event['ResourceProperties']['AccountId']
+              role_name = event['ResourceProperties']['RoleName']
+              host_tags = event['ResourceProperties']['HostTags']
+              cspm = event['ResourceProperties']['CloudSecurityPostureManagement']
+              metrics_enabled = event['ResourceProperties']['DisableMetricCollection']
```

Comment:
Another place where this is really confusing.  Since the DD API takes in `metrics_collection_enabled` I'd suggest having the CFN template param also represent "enabled" and default it to true.

Reply:
I agree. Ryan disagreed and wants the parameter to be phrased as disabling.